lib/collection/response.js

```diff
@@ -96,12 +96,265 @@ var util = require('../util'),
+
+    /**
+     * @private
+     * Regex to match non-latin characters
+     */
+    nonLatinCharMatchRegex = /[^\x20-\x7e\xa0-\xff]/g,
+
+    /**
+     * Replaces non-latin characters with '?'
+     *
+     * @private
+     * @param {string} val - The string which needs to be checked
+     */
+    getLatin = function (val) {
+        return String(val).replace(nonLatinCharMatchRegex, '?');
+    },
+
+    /**
+     * Decodes the hexcode to charCode
+     *
+     * @private
+     * @param {string} str
+     * @param {string} hex
+     * @returns {string} String with decoded hexcode values
+     */
+    decodeHexcode = function (str, hex) {
+        return String.fromCharCode(parseInt(hex, 16));
+    },
+
+    /**
+     * Decodes the given filename with given charset
+     * The supported charset are iso-8859-1 and utf-8
+     *
+     * @private
+     * @param {string} encodedFileName
+     * @param {string} charset
+     * @returns {string} Returns the decoded filename
+     */
+    decodeFileName = function (encodedFileName, charset) {
+        if (!encodedFileName) {
+            return;
+        }
+        // Decodes the hexadecimal numbers to charCode
+        var binary = encodedFileName.replace(hexCharMatchRegex, decodeHexcode),
+            value;
+        switch (charset) {
+            case iso88591:
+                value = getLatin(binary);
+                break;
+            case utf8:
+                if (!supportsBuffer) {
+                    return;
+                }
+                value = (new Buffer(binary, BINARY)).toString(UTF8);
+                break;
+            default:
+                return;
+        }
+        return value;
+    },
+
+    /**
+     * Splits the file base name and extension name
+     *
+     * @private
+     * @param {string} fileName
+     * @returns {string} Returns the decoded filename
+     */
+    splitFileName = function (fileName) {
+        var fileNameSplitMatches;
+        if (!fileName) {
+            return;
+        }
+        fileNameSplitMatches = fileNameSplitRegex.exec(fileName);
+        // Returns default file name if filename is empty in dispositionHeader
+        if (!fileNameSplitMatches || !fileNameSplitMatches[1]) {
+            return;
+        }
+        return {
+            name: fileNameSplitMatches[1],
+            ext: fileNameSplitMatches[2] || E
+        };
+    },
+
+    /**
+    * Parses Content disposition header, and returns file name and extension, a fallback of default file name
+    * and extension from content type is returned if dispositionHeader or file name is not present
+    *
+    * @private
+    * @param {String} dispositionHeader -Content-disposition Header from the response
+    * @param {string} contentType -The sanitized content type
+    * @returns {Object} Returns file base name and extension name
+    */
+    getFileNameFromDispositionHeader = function (dispositionHeader, contentType) {
+        var defaultFileName = {
```

Comment:
This helper function should not choose default. It should be done at the point where this function is consumed.